api/tools/src/main/java/org/pmiops/workbench/tools/ExportWorkspaceData.java

```diff
@@ -190,15 +142,79 @@ public CommandLineRunner run(
       final CustomMappingStrategy mappingStrategy = new CustomMappingStrategy();
       mappingStrategy.setType(WorkspaceExportRow.class);
 
-      StatefulBeanToCsv beanWriter =
-          new StatefulBeanToCsvBuilder(writer).withMappingStrategy(mappingStrategy).build();
+      try (FileWriter writer =
+          new FileWriter(opts.getOptionValue(exportFilenameOpt.getLongOpt()))) {
+        StatefulBeanToCsv beanWriter =
+            new StatefulBeanToCsvBuilder(writer).withMappingStrategy(mappingStrategy).build();
 
-      beanWriter.write(rows);
-
-      writer.close();
+        beanWriter.write(rows);
+      }
     };
   }
 
+  private WorkspaceExportRow toWorkspaceExportRow(DbWorkspace workspace) {
```

Comment:
This is a small point, but worth considering if you want to generalize to other representations. I feel like the target class WorkspaceExportRow should contain the knowledge of how the columns are formatted, (or indeed if there are columns at all), and this method should pass in raw data in its most natural form (like integers instead of strings).